### ISSUE TITLE
Width is only set after keyboard is dismissed

### ISSUE THREAD

**Issue by ristiisa** [lzhuor/auto_size_text_field#4]
![ezgif-2-d2a51b4a6e04](https://user-images.githubusercontent.com/514571/91840384-48739400-ec59-11ea-9dd5-2de8a633b63e.gif)

It seems something is broken. This issue seems similar to #3, but it seems that in that issues case the width does not increase at all.

To repro: https://github.com/lzhuor/auto_size_text_field/blob/master/example/lib/main.dart

 `Flutter (Channel dev, 1.22.0-1.0.pre, on Mac OS X 10.15.4 19E287, locale en-EE)`

**Comment by bounty1342** [comment]
Same on stable 1.20.4

**Comment by lzhuor** [comment]
Thanks @bounty1342 @ristiisa . This is known. I usually do the trick @bounty1342 mentioned. I will look into it.

Apologies, I was busy in an app release.

**Comment by grinder15** [comment]
While the current trick works, It sacrifices performance if you rebuild the unnecessary subtree of widgets, It can be optimize if you separate the `AutoSizeTextField` to another stateful widget and do rebuilds there.

```
class MyUselessStatefulWidget extends StatefulWidget {
     // constructor omitted.
     final TextEditingController controller;
     // other fields you need to customize the field
     // omitted.
}

class _MyUselessStatefulWidgetState extends State<MyUselessStatefulWidget> {
    Widget build(Buildcontext context) {
         return AutoSizeTextField(
               controller: widget.controller,
               // set the properties here from widget reference. omitted.
              onChanged: (_) => setState(() {}),
         );
     }
}
```

Better if AutoSizeTextField rebuilds itself by controller changes. By inspection of the code, I find out that the field is also a stateful widget but not a single `setState` is called there. This should be fixed.

**Comment by lzhuor** [comment]
@grinder15 good point. I will look into it this week.

**Comment by lzhuor** [comment]
Thanks @ristiisa for reporting the issue and @grinder15 for the suggestion.

I implemented @grinder15 's suggestion in commit 20a7f8f
